React/Modules/RCTDevSettings.mm

```diff
@@ -0,0 +1,572 @@
+  NSURLSessionDataTask *_liveReloadUpdateTask;
+  NSURL *_liveReloadURL;
+  BOOL _isJSLoaded;
+}
+
+@property (nonatomic, strong) Class executorClass;
+@property (nonatomic, readwrite, strong) id<RCTDevSettingsDataSource> dataSource;
+
+@end
+
+@implementation RCTDevSettings
+
+@synthesize bridge = _bridge;
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init
+{
+  // default behavior is to use NSUserDefaults
+  NSDictionary *defaultValues = @{
+    kRCTDevSettingShakeToShowDevMenu: @YES,
+  };
+  RCTDevSettingsUserDefaultsDataSource *dataSource = [[RCTDevSettingsUserDefaultsDataSource alloc] initWithDefaultValues:defaultValues];
+  return [self initWithDataSource:dataSource];
+}
+
+- (instancetype)initWithDataSource:(id<RCTDevSettingsDataSource>)dataSource
+{
+  if (self = [super init]) {
+    _dataSource = dataSource;
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(jsLoaded:)
+                                                 name:RCTJavaScriptDidLoadNotification
+                                               object:nil];
+
+    // Delay setup until after Bridge init
+    __weak RCTDevSettings *weakSelf = self;
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [weakSelf _synchronizeAllSettings];
+      [weakSelf connectPackager];
+    });
+  }
+  return self;
+}
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+- (void)invalidate
+{
+  [_liveReloadUpdateTask cancel];
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)updateSettingWithValue:(id)value forKey:(NSString *)key
+{
+  [_dataSource updateSettingWithValue:value forKey:key];
+}
+
+- (id)settingForKey:(NSString *)key
+{
+  return [_dataSource settingForKey:key];
+}
+
+- (BOOL)isRemoteDebugAvailable
+{
+  Class jsDebuggingExecutorClass = objc_lookUpClass("RCTWebSocketExecutor");
+  return (jsDebuggingExecutorClass != nil);
+}
+
+- (BOOL)isHotLoadingAvailable
+{
+  return _bridge.bundleURL && !_bridge.bundleURL.fileURL; // Only works when running from server
+}
+
+- (BOOL)isLiveReloadAvailable
+{
+  return (_liveReloadURL != nil);
+}
+
+- (BOOL)isJSCSamplingProfilerAvailable
+{
+  return JSC_JSSamplingProfilerEnabled(_bridge.jsContext.JSGlobalContextRef);
+}
+
+RCT_EXPORT_METHOD(reload)
+{
+  [_bridge reload];
+}
+
+- (void)setWebsocketExecutorName:(NSString *)websocketExecutorName
+{
+  [self updateSettingWithValue:websocketExecutorName forKey:kRCTDevSettingWebsocketExecutorName];
+}
+
+- (NSString *)websocketExecutorName
+{
+  return [self settingForKey:kRCTDevSettingWebsocketExecutorName];
+}
+
+- (void)setIsShakeToShowDevMenuEnabled:(BOOL)isShakeToShowDevMenuEnabled
+{
+  [self updateSettingWithValue:@(isShakeToShowDevMenuEnabled) forKey:kRCTDevSettingShakeToShowDevMenu];
+}
+
+- (BOOL)isShakeToShowDevMenuEnabled
+{
+  return [[self settingForKey:kRCTDevSettingShakeToShowDevMenu] boolValue];
+}
+
+RCT_EXPORT_METHOD(setIsDebuggingRemotely:(BOOL)enabled)
```

Comment:
The old DevMenu exports `debugRemotely:` to JS. In the new implementation, that old method is deprecated and wraps this method. The new method is still exposed to JS.